component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -1008,7 +1010,6 @@ protected void processEmailOTPFlow(HttpServletRequest request, HttpServletRespon
                 String ipAddress = IdentityUtil.getClientIpAddress(request);
                 context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, myToken);
                 context.setProperty(EmailOTPAuthenticatorConstants.OTP_GENERATED_TIME, System.currentTimeMillis());
-                context.setProperty(EmailOTPAuthenticatorConstants.OTP_EXPIRED, "false");
```

Comment:
Why we remove this part from here?

Reply:
The `OTP_EXPIRED` property has been set to `false` here whenever a new OTP has been generated. But we require the `OTP_EXPIRED` property in order to display proper error messages when OTP has expired. That's why I have moved this line to here (https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/153/files#diff-64287a507e2f28fd63b8a663df5180ba9e5091fc3209309f34dcaab1efdd52f6R310)
Moving the particular line of code to this place would give the same effect since the function `processAuthenticationResponse` will be triggered each time whenever an OTP has been submitted by the user.